crates/primitives/src/hardfork.rs

```diff
@@ -72,15 +56,13 @@ impl Hardfork {
 
     /// Creates a [`ForkFilter`](crate::ForkFilter) for the given hardfork.
     ///
-    /// **CAUTION**: This assumes the current hardfork's block number is the current head and uses
-    /// all known future hardforks to initialize the filter.
```

Comment:
is this no longer the case?

Reply:
The behavior is consistent with the comment, but the caution is probably not warranted because we are explicitly passing in a `ChainSpec`. Before #747, the signature was:
```rust
pub fn fork_filter(&self) -> ForkFilter
```
which is not explicit as to what future fork blocks are used to initialize the `ForkFilter`, hence the caution.

So maybe the comment should be added back, but without the `**CAUTION**`.